url0d084b4168_zhourui392_BaiduPicDown_tgz-pJ8-org_json_TestJ8/src/zz/spider/DownloadImage.java

```diff
@@ -0,0 +1,84 @@
+package zz.spider;
+
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+
+import org.nutz.img.Images;
+
+import zz.config.ZZConfig;
+import zz.http.HttpUtil;
+
+public class DownloadImage implements Runnable {
+    private String imageSrc;
+    private String imageName;
+     
+    private String mSavePath;
+     
+    public DownloadImage(String imageSrc, String savePath) {
+        this.imageSrc = imageSrc;
+        mSavePath = savePath;
+    }
+     
+    private boolean isImageExists() {
+        File dir = new File(mSavePath);
+        if(!dir.exists()) {
+            dir.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)